src/main/java/fr/inria/corese/w3c/junit/dynamic/executor/impl/RdfCanonicalEvaluationTestExecutor.java

```diff
@@ -0,0 +1,218 @@
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Executor for positive evaluation tests of RDF Canonicalization (RDFC10EvalTest).
+ * This executor performs the RDFC-1.0 canonicalization algorithm on an input RDF model
+ * and compares the resulting canonical N-Quads set with the expected canonical N-Quads set.
+ */
+public class RdfCanonicalEvaluationTestExecutor implements TestExecutor {
+
+    private static final Logger logger = LoggerFactory.getLogger(RdfCanonicalEvaluationTestExecutor.class);
+
+    // Base URL for W3C RDF Canonicalization tests
+    private static final String W3C_BASE_URL = "https://w3c.github.io/rdf-canon/tests/";
+
+    /**
+     * Constructs a new RdfCanonicalEvaluationTestExecutor.
+     */
+    public RdfCanonicalEvaluationTestExecutor() {
+        // No initialization required
+    }
+
+    /**
+     * Executes the RDF Canonicalization evaluation test case.
+     * The process involves loading the input (action) model, canonicalizing it,
+     * loading the expected (result) model, and comparing the resulting N-Quads statements.
+     *
+     * @param testCase The W3cTestCase containing the test metadata (URIs, name).
+     * @throws Exception If any error occurs during file loading, parsing, or canonicalization.
+     * @throws AssertionError If a StackOverflowError is detected during canonicalization, indicating cyclic structures.
+     */
+    @Override
+    public void execute(W3cTestCase testCase) throws Exception {
+        String testName = testCase.getName();
+        URI actionFileUri = testCase.getActionFileUri();
+        URI resultFileUri = testCase.getResultFileUri();
+
+        logger.info("Executing RDF Canonical evaluation test: {}", testName);
+
+        try {
+            // STEP 1: Resolve and load the action and result files
+            String actionFilePath = resolveAndLoadFile(actionFileUri, testName);
+            String resultFilePath = resolveAndLoadFile(resultFileUri, testName);
+
+            // STEP 2: Create and populate the action model from the input file
+            Model actionModel = RDFTestUtils.createModel();
+            RDFFormat actionFormat = RDFFormat.NQUADS;
+            RDFParser actionParser = RDFTestUtils.createParser(actionFormat, actionModel);
+
+            try (FileReader reader = new FileReader(actionFilePath)) {
+                actionParser.parse(reader);
+            } catch (IOException e) {
+                logger.error("Failed to read or parse action file: {}", actionFilePath, e);
+                throw e;
+            }
+
+            // Log the input statements
+            logger.debug("Input statements for test {}:", testName);
+            List<Statement> inputStatements = actionModel.stream().toList();
+            for (int i = 0; i < inputStatements.size(); i++) {
+                Statement stmt = inputStatements.get(i);
+                logger.debug("  [{}] {}", i, StatementUtils.toNQuad(stmt));
+            }
```

Comment:
```suggestion
            if (logger.isDebugEnabled()) {
                logger.debug("Input statements for test {}:", testName);
                List<Statement> inputStatements = actionModel.stream().toList();
                for (int i = 0; i < inputStatements.size(); i++) {
                    Statement stmt = inputStatements.get(i);
                    logger.debug("  [{}] {}", i, StatementUtils.toNQuad(stmt));
                }
            }

```

The problem is that StatementUtils.toNQuad(stmt) is called unconditionally even when debug logging is disabled, which is inefficient. You should wrap the debug logging in a conditional check.